templates/react/field-types/Timestamp/DateTimePicker/edit/render.hbs

```diff
@@ -0,0 +1,2 @@
+<label htmlFor="{{label}}">{{label}}</label>
+<input name="{{label}}" type="datetime-local" {{#if variable_to_read}} value={new Date({{variable_to_read}}!).toISOString()} {{/if}} onChange={(e) => set{{pascal_case variable_to_change}}(Math.floor(new Date(e.target.value).getTime() / 1000))} {{#if required}}required{{/if}} />
```

Comment:
It blows me away that the standards bodies have given us a `datetime-local` input that takes a format that is not supported by `Date.prototye`. Total mismatch there. Here's something I cobbled together that gets rid of errors:

```typescript
  Date.prototype.formatLocalDateForPicker = function() {
    return `${String(this.getFullYear()).padStart(4, '0')}-${String(this.getMonth() + 1).padStart(2, '0')}-${String(this.getDate()).padStart(2, '0')}T${String(this.getHours()).padStart(2, '0')}:${String(this.getMinutes()).padStart(2, '0')}:${String(this.getSeconds()).padStart(2, '0')}`;
  };                             
```

Which then lets me go:

```suggestion
<input name="{{label}}" type="datetime-local" {{#if variable_to_read}} value={new Date({{variable_to_read}}!).formatLocalDateForPicker()} {{/if}} onChange={(e) => set{{pascal_case variable_to_change}}(Math.floor(new Date(e.target.value).getTime() / 1000))} {{#if required}}required{{/if}} />
```

Is it also true that the underlying scaffolded type assumes micros? The default value I got had a five-digit year. If so, I guess it'll need to look like this:

```suggestion
<input name="{{label}}" type="datetime-local" {{#if variable_to_read}} value={new Date({{variable_to_read}}! / 1000).formatLocalDateForPicker()} {{/if}} onChange={(e) => set{{pascal_case variable_to_change}}(Math.floor(new Date(e.target.value).getTime() * 1000))} {{#if required}}required{{/if}} />
```

(see also [this line](https://github.com/holochain/scaffolding/pull/295/files#diff-811d1075d4fbff6626ca9a64c806ae7489bd6aab8097518799c80ae770b9b36cR1), which does apply that math)

Reply:
Thanks for the suggestion, I have simplified `formatLocalDateForPicker` further to this

`new Date({{variable_to_read}}! / 1000).toISOString().slice(0, 16)`

Reply:
That is simpler! As a default experience it might be confusing for end-users, though, because this gives a UTC date rather than the local date that the widget (and the user?) is expecting.